### PR TITLE
test: [M3-8809] - Add vitest workspace configuration

### PR DESCRIPTION
**Pull request by hkhalil-akamai**
## Description 📝
Adds a [Vitest workspace configuration file](https://vitest.dev/guide/workspace) ([`vitest.workspace.ts`](https://github.com/linode/manager/pull/11184/files#diff-f0ebedc84f6433817c79842f8de16366429ad939bd34d1cefdf6526fde4264f5)), allowing the running and reporting of unit tests from all packages with a single command.

This change was originally suggested in the original UI package PR: https://github.com/linode/manager/pull/11057#discussion_r1792540153

## Changes  🔄
- Adds `vitest.workspace.ts` configuration file
- Updates commands in `package.json` to run tests across all packages
- Updates a single unit test that was failing when run from workspace root

## How to test 🧪

In the repo root, run:

```
yarn test
```

### CODE REVIEW

File: packages/manager/.storybook/utils.ts
Comment (hkhalil-akamai) [inline packages/manager/.storybook/utils.ts:1]: This module was originally `$PATH`-dependent, causing it to fail when run from the repo root.

**Review by bnussman-akamai — APPROVED** [review]
This is great!

<hr />
This is probably out of scope of this PR:

Looks like `docs/development-guide/08-testing.md` needs some updating.
For example  `yarn test --watch` doesn't work. That may have been a jest thing?

We could add a  `yarn test:watch` for vitest and update the docs, but I usually just use `yarn workspace linode-manager vitest` directly when I need to use watch mode so I don't feel strongly about it

**Review by jdamore-linode — APPROVED** [review]
This is really cool, thanks @hkhalil-akamai!

**Review by mjac0bs — APPROVED** [review]
Thank you for doing this! 🥇 

Tests from both packages looking great:
Screenshot 2024-10-30 at 8 35 51 AM

(Should we add a M3 ticket for this to the Modularization epic?)

**Comment by hkhalil-akamai** [comment]
> (Should we add a M3 ticket for this to the Modularization epic?)

Added in M3-8809.

**Review by dwiley-akamai — APPROVED** [review]
I got some unit test failures locally, but everything passes in CI ✅

**Pull request merged**